icu4c/source/tools/toolutil/toolutil.cpp

```diff
@@ -218,6 +219,37 @@ uprv_fileExists(const char *file) {
 }
 #endif
 
+U_CAPI int32_t U_EXPORT2
+uprv_compareGoldenFiles(
+        const char* buffer, int32_t bufferLen,
+        const char* goldenFilePath,
+        bool overwrite) {
+
+    if (overwrite) {
+        std::ofstream ofs;
+        ofs.open(goldenFilePath);
+        ofs.write(buffer, bufferLen);
+        ofs.close();
+    }
```

Comment:
Why not return -1 after this? Surely comparing the just-written buffer with itself is unnecessary?

Reply:
Done.